drape_frontend/stylist.cpp

```diff
@@ -144,21 +154,34 @@ class Aggregator
   void Init()
   {
     m_featureLayer = m_f.GetLayer();
-    // @todo m_priorityModifier is not used, try to apply to areas to render in size order.
-    if (m_geomType == feature::GeomType::Point)
-      m_priorityModifier = (double)m_f.GetPopulation() / 7E9;
-    else
+    if (m_geomType == feature::GeomType::Area)
     {
+      // Calculate depth based on areas' sizes instead of style-set priorities.
       m2::RectD const r = m_f.GetLimitRect(m_zoomLevel);
-      m_priorityModifier = std::min(1.0, r.SizeX() * r.SizeY() * 10000.0);
+
+      // Raw areas' size range of about (1e-10, 3000) is too big, have to shrink it.
+      double const areaSize = r.SizeX() * r.SizeY();
+      // log2() of numbers <1.0 is negative, adjust it to be positive.
+      double const minLog2 = std::log2(std::numeric_limits<float>::min()); // == -126
+      double const areaSizeCompact = std::log2(areaSize) - minLog2;
```

Comment:
I used `log2()` just in a hope that binary logarithm is faster to compute :)
Actually we can use `log10()`, `log()` or `pow()` even - whatever is faster.

Reply:
BTW, you can take out this constant also into the beginning of the module like
double const kMinLog2 = std::log2(std::numeric_limits<float>::min()); // == -126

Reply:
Done